### PR TITLE
Merge eslint-config-hapi and rules

### PR DESCRIPTION
**Pull request by kanongil**
This PR aims to simplify the linting setup by merging the config from `@hapi/eslint-config-hapi`. This allows the config & plugin to be installed through a single dependency, without `peerDependency` shenanigans. It should also simplify future maintenance.

This is not a breaking change, as the existing functionality will continue to work.

If this is merged, `@hapi/eslint-config-hapi` should be changed to just extend `plugin:@hapi/hapi/recommended` and/or be deprecated. Additionally, `@hapi/lab` should be updated extend from `plugin:@hapi/hapi/recommended `, and remove the `@hapi/eslint-config-hapi` dependency.

Note: Ideally the plugin should also be renamed to `@hapi/eslint-plugin`. This would simplify the extend line to `plugin:@hapi/recommended`, but it is a breaking change.

### CODE REVIEW

**Comment by cjihrig** [comment]
Simplifying the linting setup sounds good to me. It looks like none of the hapi specific ESLint rules are depended on by anyone else. Can we just pull those five repos into here as well?

**Comment by kanongil** [comment]
Yes, I wanted to do that but limited it to this for an initial pass. I can add another commit with the rules as well.

**Comment by kanongil** [comment]
PR updated with merged rules.

The self-dependency seems to mess up the windows builds, causing CI failures. I can fix it by clearing the travis cache and rerunning the test. I suspect caching will need to be disabled to work consistently.

FYI I also have a local branch which updates the rules implementation to better match modern eslint rules.

File: API.md [hunk 1]
```diff
@@ -1,25 +1,169 @@
...
 
+Prevents arrow functions that implicitly create additional arrow functions.
+
+This rule prevents the pattern () => () => () => ...;.
+
+Functions can still be returned by arrow functions whose bodies use curly braces and an explicit return.
+
+This rule does not accept any configuration options.
+
+## Config
```
Comment (cjihrig) [inline API.md:77]: Maybe this should just link to the config file. This is duplicating a lot, and could easily get out of sync.

Comment (kanongil) [reply API.md:77]: This PR just tries to merge. Feel free to change this in a subsequent commit.

**Comment by kanongil** [comment]
FYI, I have another patch lined up to update the rules to look more like current eslint plugins.

**Comment by kanongil** [comment]
As I alluded, you might also want to do another followup patch to change the package name.

**Comment by cjihrig** [comment]
@kanongil do you have merge permissions? If so, are you comfortable landing this?

**Comment by kanongil** [comment]
The merge has been merged, and a followup PR posted for review in #22.

**Pull request merged**